packages/starrating/src/react/star.js

```diff
@@ -0,0 +1,146 @@
+import * as glamor from 'glamor'
+import PropTypes from 'prop-types'
+import React from 'react'
+
+import Icon from '@pluralsight/ps-design-system-icon/react'
+import { withTheme } from '@pluralsight/ps-design-system-theme/react'
+
+import css, { BASE_CLASSNAME } from '../css'
+
+const APPEARANCES = {
+  empty: 'empty',
+  half: 'half',
+  full: 'full'
+}
+
+const styles = {
+  star: props =>
+    glamor.css(
+      css[`${BASE_CLASSNAME}__star`],
+      css[`${BASE_CLASSNAME}__star--theme-${props.themeName}`],
+      props.active && css[`${BASE_CLASSNAME}__star--active`],
+      props.interactive && css[`${BASE_CLASSNAME}__star--interactive`]
+    ),
+  halfStarSecondary: props =>
+    glamor.css(
+      css[`${BASE_CLASSNAME}__star__half__secondary`],
+      css[`${BASE_CLASSNAME}__star__half__secondary--theme-${props.themeName}`]
+    )
+}
+
+const HalfStarIcon = props => {
+  const { themeName, ...filteredProps } = props
+
+  return (
+    <Icon {...filteredProps}>
+      <svg
+        role="img"
+        aria-label="star half icon"
+        viewBox="0 0 24 24"
+        xmlns="http://www.w3.org/2000/svg"
+      >
+        <defs>
+          <symbol id="star">
```

Comment:
Do these ids need namespaced to the DS and the component? Might be good to be a good citizen inside others' apps.

Reply:
let me look into it. svg defs might do something like shadow dom but i'm not sure. i think namespacing would be good